infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -273,26 +295,34 @@ public void read(CorfuPayloadMsg<ReadRequest> msg, ChannelHandlerContext ctx, IS
             log.trace("read: {}, cacheable: {}", msg.getPayload().getAddresses(), cacheable);
         }
 
-        ReadResponse rr = new ReadResponse();
+        Lock lock = resetLock.readLock();
+        lock.lock();
 
-        for (long address : msg.getPayload().getAddresses()) {
-            try {
-                ILogData logData = dataCache.get(address, cacheable);
-                if (logData == null) {
-                    rr.put(address, LogData.getEmpty(address));
-                } else {
-                    rr.put(address, (LogData) logData);
+        try {
+
+            ReadResponse rr = new ReadResponse();
+
```

Comment:
The epoch check for reads was dropped from the last patch. Please re-add it.

Reply:
Forgot to move it from `processRequest`, my bad.
Added that check into processRequest for
```
 READ_REQUEST:
 KNOWN_ADDRESS_REQUEST:
 COMPACT_REQUEST:
```

Reply:
Unfortunately, the change break tests https://github.com/CorfuDB/CorfuDB/pull/2661#issuecomment-667404526